packages/core/__tests__/common/getFilesList.test.ts

```diff
@@ -273,6 +273,41 @@ it('should return files for project root with omitting gitignore', async () => {
   )
 })
 
+it('should use .gitignore only within the current repository', async () => {
+  mockFs({
+    [root]: {
+      '.git': {},
+      project: {
+        'fileA.ts': 'content',
+        '.gitignore': 'packageA/*',
+        packageA: {
+          '.git': {},
+          '.env': 'content',
+          '.gitignore': `
+            .env
+          `,
+          src: {
+            'fileE.json': '',
+          },
+        },
+      },
+    },
+  })
+
+  const filesList = removeCwd(
+    await getFilesList({
+      searchRoot: toPlatformSpecificPath(`${root}/project/packageA`),
+    }),
+  )
+  mockFs.restore()
+
+  expect(filesList.sort()).toMatchObject(
+    [`${root}/project/packageA/src/fileE.json`]
+      .sort()
+      .map(toPlatformSpecificPath),
+  )
```

Comment:
Currently, this returns an empty array because the implementation will look at the `.gitignore` on line 282